lily4178993/awesome-books-2
# Features

# Awesome Books: Final

This PR is raised to implement the following features according to the project [requirements](https://github.com/microverseinc/curriculum-javascript/blob/main/books/books_with_es6.md).  The goal is to make the code more organized by using modules and also practice the ES6 syntax that I have learned.

## Features implemented 👌
The following features have been implemented:

- [x] A Navigation bar with nav menu items.
- [x] When  any of the links, are clicked the main area changes to display only one of the three content sections.
- [x] Used only plain JavaScript and CSS.

**Added Three content sections:**
- [x] Books list.
- [x] Add book form.
- [x] Contact info.


In addition, the following functionalities from the first [implementation](https://github.com/microverseinc/curriculum-javascript/blob/main/books/m1_plain_js_objects_v1_1.md) have been added:
*   When a user clicks the "Remove" button: The correct book is removed from the collection.
*   Data is preserved in the browser's memory by using local storage.
* When a user fills in the form title and author and presses the add button, a book is added to the list.


## Project requirements
The following requirements have also been fulfilled:
**HTML & CSS:**
* Used semantic HTML tags.
* Used CSS selectors correctly.
* Followed the list of [best practices for HTML & CSS](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/html_css_best_practices.md).

**Javascript:**
* Used JavaScript to manipulate DOM elements.
* Used JavaScript events.
* Used objects to store and access data.
* Create a separate file for each module in the modules directory.
* Imported modules into the index.js file.
* Refactored all functions to arrow functions.
* Followed the list of [best practices for JavaScript](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md).

**Linters:**
* There are [no linter errors](https://github.com/microverseinc/linters-config).

**Documentation:**
* Used correct [GitHub flow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/github_flow.md).
* Documented our work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).


  
## How should it be tested ? 🧪 

Clone the repository
```
git clone https://github.com/lily4178993/awesome-books-2
```
Navigate to the folder
```
cd awesome-books-2
```
Checkout the branch
```
git checkout features
```

**Awesome books** is hosted on GitHub pages, so if you want to check this project online, please 😊 [Click here](https://lily4178993.github.io/awesome-books-2/)

_**NOTA BENE: 💀 Unable to fix webhint error!**_

Please kindly review my implementation and suggest any changes and improvements.

I look forward to your feedback.
Thanks for reviewing my PR 😊
